### PR TITLE
Update secureboot and mokutil test case

### PR DESCRIPTION
**Pull request by mloviska**
Test objects:
- mokutil:
    - query enrolled certificates
    - import and remove mock certificate
    - check secureboot status

- sbsignutil
    - verify shim certificate
    - list certificates

- update-bootloader reinit and shim install

- Verification runs:
  * [sle-15-SP3-JeOS-for-MS-HyperV-x86_64](http://kepler.suse.cz/tests/4897#)
  * [sle-15-SP3-JeOS-for-kvm-and-xen-x86_64-](http://kepler.suse.cz/tests/4893#step/verify_efi_mok/1)
  * [sle-15-SP2-JeOS-for-kvm-and-xen-QR-x86_64-Build15.87-jeos-main@uefi-virtio-vga](http://kepler.suse.cz/tests/4892#step/verify_efi_mok/1)
  * [opensuse-15.2-JeOS-for-kvm-and-xen-x86_64](http://kepler.suse.cz/tests/4888)
  * [opensuse-Tumbleweed-JeOS-for-kvm-and-xen](http://kepler.suse.cz/tests/4886#step/verify_efi_mok/1)
  * [opensuse-15.3-JeOS-for-kvm-and-xen-x86_64](http://kepler.suse.cz/tests/4887#step/verify_efi_mok/1)

### CODE REVIEW

**Review by asmorodskyi — APPROVED** [review]
LGTM

**Comment by pdostal** [comment]
Did you test this on XEN?

**Comment by mloviska** [comment]
> Did you test this on XEN?

Yup, in the past. AFAICT xen + uefi is not really supported by SUSE. But should be doable in the future

**Review by pdostal — APPROVED** [review]

File: tests/console/verify_efi_mok.pm [hunk 1]
```diff
@@ -121,56 +112,49 @@ sub check_mok {
...
         svirt_hyperv => 'scsi'
     };
-
-    return $blk_dev_driver->{$backend};
-}
-
-sub get_esp_from_parted {
-    my @parted_data = split /\n/, script_output 'parted --list --machine --script';
+    my $virtualized_block_type = sub {
+        return $blk_dev_driver->{${\(join '_', grep /$_/, get_required_var 'BACKEND', get_var 'VIRSH_VMM_FAMILY')}};
```
Comment (cfconrad) [inline tests/console/verify_efi_mok.pm:145]: Why do you need this interpolation here?
```suggestion
        return $blk_dev_driver->{join ('_', grep (/$_/, get_required_var 'BACKEND', get_var 'VIRSH_VMM_FAMILY'))};
```

Comment (cfconrad) [reply tests/console/verify_efi_mok.pm]: Are you sure about the change in the `grep()`.  If I understood it right you try to eleiminate `undef` from the list before.
So it is:
```bash
FOOF__BARclemix@linux-7k5u:~> perl -e 'print join("_", grep(/$_/, "FOOF", undef, "BAR")) . $/;'
FOOF__BAR
```
vs
```
clemix@linux-7k5u:~> perl -e 'print join("_", grep {$_} "FOOF", undef, "BAR") . $/;'
FOOF_BAR
```

Comment (cfconrad) [reply tests/console/verify_efi_mok.pm]: Actually I don't understand why you changed this function, can you help me?

Comment (mloviska) [reply tests/console/verify_efi_mok.pm]: > Are you sure about the change in the `grep()`. If I understood it right you try to eleiminate `undef` from the list before.
> So it is:
> 
> ```shell
> FOOF__BARclemix@linux-7k5u:~> perl -e 'print join("_", grep(/$_/, "FOOF", undef, "BAR")) . $/;'
> FOOF__BAR
> ```
> 
> vs
> 
> ```shell
> clemix@linux-7k5u:~> perl -e 'print join("_", grep {$_} "FOOF", undef, "BAR") . $/;'
> FOOF_BAR
> ```

Actually `undef` yields warnings, so the `VIRSH_VMM_FAMILY` should return empty string, I will update that.
The thing is that each hypervisor has a different block device type. Therefore, I have this lookup table for it.
```
 my $blk_dev_driver = {
        qemu         => 'virtblk',
        svirt_xen    => 'xvd',
        svirt_hyperv => 'scsi'
    };
```my $a = 'qemu';
my $b = '';
my $blk_dev_driver = {
        qemu  => 'virtblk',
        svirt_xen => 'xvd',
        svirt_hyperv => 'scsi'
};
print "Testing \$a == $a and \$b == $b\n";
my $testfce = sub { return $blk_dev_driver->{join '_', grep /$_/, $a, $b}};
```

Got me
```
Testing $a == qemu and $b ==
T1: virtblk
```

Comment (mloviska) [reply tests/console/verify_efi_mok.pm]: > ```shell
> perl -e 'print join("_", grep(/$_/, "FOOF", undef, "BAR")) . $/;'
> ```

Weird, I thought that `grep BLOCK LIST` or `grep EXPR,LIST` are equivalent forms. Let me do more tests. Thanks for the point!

Comment (mloviska) [reply tests/console/verify_efi_mok.pm]:
```
Another join test
Testing $a == qemu and $b ==
T1 EXPR: qemu
T1 BLOCK: qemu
Testing $a == qemu and $b ==  and $c == BAR
T1.3 EXPR: qemu_BAR
T1.3 BLOCK: qemu_BAR
```

```print "Another join test\n";
my $a = 'qemu';
my $b = '';
my $c = 'BAR';

print "Testing \$a == $a and \$b == $b\n";
my $testfce = sub { join '_', grep /$_/, $a, $b};
print "T1 EXPR: ${\$testfce->()}\n";
$testfce = sub { join '_', grep {$_} $a, $b};
print "T1 BLOCK: ${\$testfce->()}\n";
print "Testing \$a == $a and \$b == $b and \$c == $c\n";
$testfce = sub { join '_', grep /$_/, $a, $b, $c};
print "T1.3 EXPR: ${\$testfce->()}\n";
$testfce = sub { join '_', grep {$_} $a, $b, $c};
print "T1.3 BLOCK: ${\$testfce->()}\n";
```

Comment (mloviska) [reply tests/console/verify_efi_mok.pm]:
```
Testing $a == qemu and $b == undef and $c == BAR
Use of uninitialized value $_ in regexp compilation at grep_cap_test.pl line 69.
Use of uninitialized value $_ in pattern match (m//) at grep_cap_test.pl line 69.
T2.3 EXPR: qemu_BAR
T2.3 BLOCK: qemu_BAR
```

Comment (cfconrad) [reply tests/console/verify_efi_mok.pm]: Don't understand, I get different result. 
https://onlinegdb.com/rJoyI8buu

Comment (cfconrad) [reply tests/console/verify_efi_mok.pm]: The `for` loop is a good choice!

**Pull request merged**